packages/picasso.js/src/core/chart-components/ref-line/lines-and-labels.js

```diff
@@ -61,27 +151,44 @@ export function alignmentToNumber(align) {
  * @param {object[]} items - Array of all items (for collision detection)
  * @ignore
  */
-export function createLineWithLabel({ chart, blueprint, renderer, p, settings, items }) {
+export function createLineWithLabel({ chart, blueprint, renderer, p, settings, items, slopeLine }) {
   let doesNotCollide = true;
   let line = false;
+  let slope = false;
   let rect = false;
   let label = false;
   let value = false;
   let style = extend(true, {}, settings.style.line, p.line || {});
+  let slopeStyle = extend(true, refLabelDefaultSettings(), settings.style.label || {}, { fill: style.stroke });
 
   // Use the transposer to handle actual positioning
-  line = blueprint.processItem({
-    type: 'line',
-    x1: p.position,
-    y1: 0,
-    x2: p.position,
-    y2: 1,
-    stroke: style.stroke || 'black',
-    strokeWidth: style.strokeWidth || 1,
-    strokeDasharray: style.strokeDasharray,
-    flipXY: p.flipXY || false, // This flips individual points (Y-lines)
-    value: p.valueInfo ? p.valueInfo.id : p.value,
-  });
+  if (slopeLine) {
+    slope = blueprint.processItem({
```

Comment:
I suggest you use the line variable for this as well, no need to have a specific slope variable.
That way you do not have to have the if statement further down:

```js
if (line) {
    items.push(line);
  } else if (slopeLine) {
    // push slope line
    items.push(slope);
```

Reply:
I don't see any value in differentiating the two - both are lines. It just adds yet another variable and extra conditional checks. So I think the code gets simpler by having one variable for the line no matter if it's sloped or not.